src/string.cr

```diff
@@ -3969,6 +3969,10 @@ class String
     false
   end
 
+  def starts_with?(re : Regex)
+    !!($~ = re.match_at_byte_index(self, 0, Regex::Options::ANCHORED))
```

Comment:
That's a fairly confusing definition, how about:
```cr
def starts_with?(re : Regex)
  match_data = $~ = re.match_at_byte_index(self, 0, Regex::Options::ANCHORED)

  match_data != nil
end
```

Reply:
Your code has another confusability. It seems `match_data` is redundant and last statement can be replaced with `$~ != nil`, however it cannot. It is hard to explain this reason, so I cannot your suggestion. Please approve.

Reply:
I disagree.

Reply:
Why?

Reply:
I just think that this way is easier to understand. Perhaps:
```cr
match_data = re.match_at_byte_index(self, 0, Regex::Options::ANCHORED)

$~ = match_data
match_data != nil
```

Reply:
This is such a small method that I think the original code is fine.